rest/src/main/java/org/jboss/pnc/rest/provider/BuildRecordProvider.java

```diff
@@ -660,63 +696,76 @@ public BuildRecordRest getLatestBuildRecord(Integer configId, boolean executedOn
+
+        logger.debug(
+                "Get running and completed build records by pageIndex: {}, pageSize: {}, sort: {}, orFindByBuildConfigurationName: {}, andFindByBuildConfigurationName: {}, rsqlQueries: {}",
+                pageIndex, pageSize, sort, orFindByBuildConfigurationName, andFindByBuildConfigurationName, rsqlQueries);
+
+        // Combine predicates of both search query and the additional filters (e.g.userId, buildConfigurationId)
         List<Predicate<BuildRecord>> dbAndPredicatesList = Arrays.stream(rsqlQueries)
                 .filter(rsqlString -> rsqlString != null && !rsqlString.isEmpty())
                 .map(p -> rsqlPredicateProducer.getPredicate(BuildRecord.class, p))
                 .collect(Collectors.toList());
         List<Predicate<BuildRecord>> dbOrPredicateList = new ArrayList<>();
 
+        // Combine rsqlQueries AND-ing all the single queries
         String combinedQueries = combineRsqlQueriesMatchAll(rsqlQueries);
-        if (!StringUtils.isEmpty(orFindByBuildConfigurationName)) {
-            //add steam condition
+
+        // An AND condition (andFindByBuildConfigurationName) gets higher precedence over an OR condition
+        // (orFindByBuildConfigurationName). If specified both, only AND is applied.
+
+        if (!StringUtils.isEmpty(andFindByBuildConfigurationName)) {
+            // add steam condition
+            StringBuilder sb = new StringBuilder(";(buildConfigurationName==").append(andFindByBuildConfigurationName).append(")");
+
             if (StringUtils.isEmpty(combinedQueries)) {
-                combinedQueries = "(buildConfigurationName=like=" + orFindByBuildConfigurationName + ")";
+                combinedQueries = sb.deleteCharAt(0).toString();
             } else {
-                combinedQueries = "(" + combinedQueries + "),(buildConfigurationName=like=" + orFindByBuildConfigurationName + ")";
+                combinedQueries = sb.insert(0, ")").insert(0, combinedQueries).insert(0, "(").toString();
             }
 
-            //add DB predicate
-            List<BuildConfigurationAudited> buildConfigurationAuditeds = buildConfigurationAuditedRepository
-                    .searchForBuildConfigurationName(orFindByBuildConfigurationName);
-            if (!buildConfigurationAuditeds.isEmpty()) {
-                dbOrPredicateList.add(
-                        BuildRecordPredicates.withBuildConfigurationIdRev(
-                                buildConfigurationAuditeds.stream().map(bca -> bca.getIdRev()).collect(Collectors.toList())
-                        )
-                );
+            // add DB predicate
+            List<IdRev> buildConfigurationAuditedIdRevs = buildConfigurationAuditedRepository
+                    .searchIdRevForBuildConfigurationName(andFindByBuildConfigurationName);
```

Comment:
The old code returned a list of BuildConfigurationAudited from their corresponding BuildConfiguration name, creating the BuildConfigurationAudited objects with an empty default BuildRecords list. It then used only the list of IdRev of the returned BuildConfigurationAudited list, to finally populate the BuildRecordPredicates list.

The new code returns directly the list of IdRev to be used to populate the BuildRecordPredicates list.